asdf/commands/edit.py

```diff
@@ -0,0 +1,529 @@
+    schema.validate(tree, None) # Failure raises and exception.
+
+    return yaml_content
+
+def edit_func ( fname, oname ) :
+    """
+    Creates a YAML file from an ASDF file.  The YAML file will contain only the
+    YAML from the ASDF file.  The YAML text will be written to a YAML text file
+    in the same, so from 'example.asdf' the file 'example.yaml' will be created.
+
+    Parameters
+    ----------
+    fname : The input ASDF file name.
+    oname : The output YAML file name.
+    """
+    if not is_validate_asdf_path(fname) :
+        return False
+
+    # Validate input file is an ASDF file.
+    fd, asdf_text = open_and_validate_asdf(fname)
+
+    # Read and validate the YAML of an ASDF file.
+    yaml_text = read_and_validate_yaml(fd,fname)
+
+    # Open a YAML file for the ASDF YAML.
+    if not is_yaml_file(oname) : 
+        # Raise an exception
+        print(f"Error: '{oname}' must have '.yaml' extension.")
+        sys.exit(1)
+
+    # Write the YAML for the original ASDF file.
+    with open(oname,"wb") as ofd :
+        ofd.write(asdf_text)
+        ofd.write(yaml_text)
+
+    # Output message to user.
+    delim = '*' * 70
+    print(f"\n{delim}")
+    print("ASDF formatting and YAML schema validated.") 
+    print(f"The text portion of '{fname}' is written to:")
+    print(f"    '{oname}'")
+    print(f"The file '{oname}' can be edited using your favorite text editor.")
+    print("The edited text can then be saved to the ASDF file of your choice")
+    print("using 'asdftool edit -s -f <edited text file> -o <ASDF file>.")
+    print('-' * 70)
+    print("Note: This is meant to be a lightweight text editing tool of")
+    print("      ASDF .If the edited text is larger than the YAML portion")
+    print("      of the ASDF file to be written to, the edits may not be")
+    print("      able to saved.")
+    print(f"{delim}\n")
+
+    return
+
+def buffer_edited_text ( edited_text, orig_text ) :
+    """ There is more text in the original ASDF file than in the edited text,
+    so we will buffer the edited text with spaces.
+    """ 
+    diff = len(orig_text) - len(edited_text)
+    if diff<1 :
+        print("Error: shouldn't be here.")
+        sys.exit(1)
+
+    wdelim = b'\r\n...\r\n'
+    ldelim = b'\n...\n'
+    if edited_text[-len(wdelim):]==wdelim :
+        delim = wdelim
+    elif edited_text[-len(ldelim):]==ldelim :
+        delim = ldelim
+    else:
+        # Mabye raise exception
+        print("Unrecognized YAML delimiter ending the YAML text.")
+        print(f"It should be {wdelim} or {ldelim}, but the")
+        print(f"last {len(wdelim)} bytes are {edited_text[-len(wdelim):]}.")
+        sys.exit(1)
+
+    # May not be correct.  If on Windows use '\r\n'.
+    buffered_text = edited_text[:-len(delim)] + b'\n' + b' '*(diff-1) + delim
+    return buffered_text, diff-1 
+
+
+def add_buffer_to_new_text ( edited_text, buffer_size ) :
+    """ Adds buffer to edited text.
+    """
+    wdelim = b'\r\n...\r\n'
```

Comment:
Can we have the user edit the YAML content between the delimiters, then add them back automatically?

Reply:
We discussed this over email and agreed to keep the entire YAML document editable.